### PR TITLE
Sentence structure changes and ISentenceFilter

### PR DESCRIPTION
**Pull request by suflors**
Changes the structure of Sentence and the OIDs inside of it, adds interface IMarkovFilter featuring basic filtering options that will be available in any implementation.

### CODE REVIEW

File: AtelierTomato.Markov/AtelierTomato.Markov.Data/Model/OIDPattern.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
+﻿using System.Text;
+using System.Text.RegularExpressions;
+
+namespace AtelierTomato.Markov.Data.Model
+{
+	public class OIDPattern
+	{
+		public static Regex Generate(IEnumerable<string> fields)
```
Comment (aliceif) [inline AtelierTomato.Markov/AtelierTomato.Markov.Data/Model/OIDPattern.cs]: If you need indexes, take a type that has indexes (i think that is IList or ICollection), that way you don't need to use ElementAt which is a method that should ideally never be used

File: AtelierTomato.Markov/AtelierTomato.Markov.Data.Test/MultiParserTest.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+namespace AtelierTomato.Markov.Data.Test
+{
+	public class MultiParserTest
+	{
+		[Fact]
+		public void DiscordIObjectOIDParseTest()
+		{
+			MultiParser<IObjectOID> oidParser = new([new BookObjectOIDParser(), new InvalidObjectOIDParser(), new DiscordObjectOIDParser()]);
+			IObjectOID OID = oidParser.Parse("Discord:discord.com:1253189664655806606:1253189664655806610");
+			OID.Should().BeEquivalentTo(DiscordObjectOID.Parse("Discord:discord.com:1253189664655806606:1253189664655806610"));
```
Comment (aliceif) [inline AtelierTomato.Markov/AtelierTomato.Markov.Data.Test/MultiParserTest.cs:16]: I would also check for expected return type here (that is, DiscordObjectOid), same in equivalent tests

File: AtelierTomato.Markov/AtelierTomato.Markov.Generation/GenerateMarkovSentence.cs [hunk 3]
```diff
@@ -4,31 +4,21 @@
...
-				throw new Exception("Couldn't query any messages.");
-			}
+			Sentence? sentence = await GetFirstSentence(filter) ?? throw new Exception("Couldn't query any messages.");
 			string firstWord = sentence.Text.Substring(0, sentence.Text.IndexOf(' '));
 			var tokenizedSentence = new List<string> { firstWord };
 			var prevList = tokenizedSentence;
 
 			// Tracks the IDs of previously used sentences so that we don't recreate an existing sentence or ping pong between two sentences.
-			var prevIDs = new List<Guid> { sentence.ID };
+			var prevIDs = new List<string> { sentence.OID.ToString() };
```
Comment (aliceif) [inline AtelierTomato.Markov/AtelierTomato.Markov.Generation/GenerateMarkovSentence.cs:21]: Can't we make this List\<IObjectOID\>?

**Review by aliceif — COMMENTED** [review]
Mildly unsure about those empty parser classes, I feel like we could maybe find a way to live without them - or make them be something real and not just wrappers

File: AtelierTomato.Markov/AtelierTomato.Markov.Generation/GenerateMarkovSentence.cs (hunk 3 above)
Comment (suflors) [reply AtelierTomato.Markov/AtelierTomato.Markov.Generation/GenerateMarkovSentence.cs:21]: Will implement in SentenceAccess branch, not a change for this pull request.

**Review by aliceif — APPROVED** [review]
should be ok

**Pull request merged**